packages/components/src/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -0,0 +1,500 @@
+                const {parentValue} = alias;
+                let value = parentValue.value as string;
+                if (parentValue === NEW_SAMPLE_SET_OPTION)
+                    value = SAMPLE_SET_IMPORT_PREFIX + name;
+
+                aliases[alias.alias] = value;
+            });
+        }
+
+        return Map<string,string>(aliases);
+    };
+
+    onFieldChange = (newModel: SampleTypeModel) => {
+        this.setState(()=>({model:newModel}));
+    };
+
+    updateAliasValue = (id:string, field: string, newValue: any): IParentAlias => {
+        const {model} = this.state;
+        const {parentAliases} = model;
+        return {
+            ...parentAliases.get(id),
+            isDupe: false,             //Clear error because of change
+            [field]: newValue,
+        } as IParentAlias;
+    };
+
+    parentAliasChange = (id:string, field: string, newValue: any) => {
+        const {model} = this.state;
+        const {parentAliases} = model;
+        const changedAlias = this.updateAliasValue(id, field, newValue);
+
+        const newAliases = parentAliases.set(id, changedAlias);
+        const newModel = model.merge({parentAliases: newAliases}) as SampleTypeModel;
+        this.setState(() => ({model: newModel}));
+    };
+
+    updateDupes = (id: string):void => {
+        const {model} = this.state;
+        if (!model) {
+            return;
+        }
+
+        const {parentAliases} = model;
+        const dupes = model.getDuplicateAlias();
+        let newAliases = OrderedMap<string, IParentAlias>();
+        parentAliases.forEach((alias: IParentAlias) => {
+            const isDupe = dupes && dupes.has(alias.id);
+            let changedAlias = alias;
+            if (isDupe !== alias.isDupe) {
+                changedAlias = this.updateAliasValue(alias.id, 'isDupe', isDupe);
+            }
+
+            if (alias.id === id) {
+                changedAlias = {
+                    ...changedAlias,
+                    ignoreAliasError: false
+                };
+            }
+
+            newAliases = newAliases.set(alias.id, changedAlias);
+        });
+
+        const newModel = model.merge({parentAliases: newAliases}) as SampleTypeModel;
+        this.setState(() => ({model: newModel}));
+    };
+
+    addParentAlias = (id:string, newAlias: IParentAlias): void => {
+        const {model} = this.state;
+        let {parentAliases} = model;
+        const newModel = model.merge({parentAliases:parentAliases.set(id, newAlias)}) as SampleTypeModel;
+        this.setState(() => ({model: newModel}));
+    };
+
+    removeParentAlias = (id:string) => {
+        const {model} = this.state;
+        let {parentAliases} = model;
+        const aliases = parentAliases.delete(id);
+        const newModel = model.set('parentAliases', aliases) as SampleTypeModel;
+        this.setState(() => ({model: newModel}));
+    };
+
+    //TODO REVIEW recommendations on where to put this that is shareable?
```

Comment:
Yeah, I have been looking into this as well for the other designers as they all have this same block of code. I have locally been playing around with either an HOC or using the Context API. I'll try addressing this after your merge and make that changes / refactor across the designers.